### ISSUE TITLE
[SUGGESTION] - Change how evidence saved

### ISSUE THREAD

**Issue by CloudTheWolf** [teddythedev/legacy-mdt-issue-tracker#9]
**Summary:**
Write a summary of what your suggestion is about.

Decouple saving evidence from saving reports / incidence so that so that `evidencegeneric` or `evidenceguns` get updated when Evidence is Added or Removed.

Eg When I fill in the evidence from and click "Add Evidence" or I delete Evidence from a report this calls an API that will update either the `evidencegeneric` or `evidenceguns` table as required, rather than doing this when the overall report is saved.

**Reason:**
Why would your suggestion be a good change for the MDT?

This will prevent evidence being lost when someone forgets to reload before updating something in the incident/report

**Comment by teddythedev** [comment]
Yeah this will come with the rewrite of incidents and saving more often. However, evidence technically isn't lost if someone saves over. The evidence gets saved it just does not have an attached incident or report anymore
